docs/security/gpg-key-for-ssh-authentication.md

```diff
@@ -0,0 +1,477 @@
+    
+    pub  4096R/71735D23  created: 2016-04-09  expires: 2017-04-09  usage: SC  
+                         trust: ultimate      validity: ultimate
+    sub  4096R/693C5635  created: 2016-04-09  expires: 2017-04-09  usage: E   
+    sub  4096R/48B9C23C  created: 2016-04-09  expires: 2017-04-09  usage: A   
+    [ultimate] (1). Your Name (Test for GPG -> SSH) <you@yoursite.net>
+
+### Securing Your GPG Key
+
+{: .caution}
+>
+> If you fail to back up or otherwise secure your key, any hardware failure will lead to you being unable to access your Linode with this key. If you lock out password access through SSH, this may be disastrous.
+
+You should always have a backup of your private key in case something goes wrong and you end up locked out of everything. This private key, along with the instructions in this guide, will be enough to get your setup working again if you need to start afresh on a new computer.
+
+1. Backup your `~/.gnupg` folder by typing:
+
+        mv ~/.gnupg/ /Volumes/USB_DEVICE/.gnupg/
+
+    (Assuming you have a storage device mounted at `/Volumes/USB_DEVICE/`)
+
+2. Backup your private key by typing:
+
+        gpg2 -a --export-secret-key <key id> >> /Volumes/USB_DEVICE/<key id>.master.key
+
+3. Backup your subkeys by typing:
+
+        gpg2 -a --export-secret-subkeys <key id> >> /Volumes/USB_DEVICE/<key id>.sub.key
+        
+If something bad happens, you can re-import your keys by overwriting the `~/.gnupg` directory with your copy, and using:
+
+    gpg2 --allow-secret-key-import --import <key file>
+
+Where `<key file>` is replaced with the location of each of your files.
+
+### Export Your Public Key
+
+If you're working on a VM/offline machine, you'll also need to export your public key to be reimported later:
+
+    gpg2 -a --export <key id> >> /Volumes/USB_DEVICE/<key id>.public.key
+    
+You can reimport it with the ever-handy `gpg2 --import <key file>` command.
+
+## Move Your Key to a Smartcard/YubiKey (Optional)
+
+{: .note}
+>
+> If you're using a brand new YubiKey, you'll need to enable OpenPGP Card / CCID Mode first.
+
+### Secure Your Card
+It is assumed that you have configured your card/YubiKey's (herein referred to as 'GPG device') owner information. You will need to secure your card before you start (or at least it's highly recommended).
+
+{: .note}
+>
+> Some of these commands may ask for a PIN or Admin PIN. The default PIN is usually `123456`, and the default Admin PIN is usually `12345678`. If these don't work, contact the manufacturer or ask around.
+
+1. Plug in the device and type:
+
+        gpg2 --card-edit
+
+2. Enable admin commands by:
+
+        admin
+
+3. Enter the password change menu:
+
+        passwd
+
+4. Change the password to your device by selecting `1 - change PIN`. This will be required every time you want to access your GPG key (i.e. every time you authenticate with SSH).
```

Comment:
Also, on my Yubikey nano (once in GPG mode as indicated on L269) I had to run option `2` to unblock the pin, which then prompted me to change it.